firebase-vertexai/src/main/kotlin/com/google/firebase/vertexai/type/LiveGenerationConfig.kt

```diff
@@ -0,0 +1,215 @@
+import kotlinx.serialization.Serializable
+
+/**
+ * Configuration parameters to use for content generation.
+ *
+ * @property temperature A parameter controlling the degree of randomness in token selection. A
+ * temperature of 0 means that the highest probability tokens are always selected. In this case,
+ * responses for a given prompt are mostly deterministic, but a small amount of variation is still
+ * possible.
+ *
+ * @property topK The `topK` parameter changes how the model selects tokens for output. A `topK` of
+ * 1 means the selected token is the most probable among all the tokens in the model's vocabulary,
+ * while a `topK` of 3 means that the next token is selected from among the 3 most probable using
+ * the `temperature`. For each token selection step, the `topK` tokens with the highest
+ * probabilities are sampled. Tokens are then further filtered based on `topP` with the final token
+ * selected using `temperature` sampling. Defaults to 40 if unspecified.
+ *
+ * @property topP The `topP` parameter changes how the model selects tokens for output. Tokens are
+ * selected from the most to least probable until the sum of their probabilities equals the `topP`
+ * value. For example, if tokens A, B, and C have probabilities of 0.3, 0.2, and 0.1 respectively
+ * and the topP value is 0.5, then the model will select either A or B as the next token by using
+ * the `temperature` and exclude C as a candidate. Defaults to 0.95 if unset.
+ *
+ * @property candidateCount The maximum number of generated response messages to return. This value
+ * must be between [1, 8], inclusive. If unset, this will default to 1.
+ *
+ * - Note: Only unique candidates are returned. Higher temperatures are more likely to produce
+ * unique candidates. Setting `temperature` to 0 will always produce exactly one candidate
+ * regardless of the `candidateCount`.
+ *
+ * @property presencePenalty Positive penalties.
+ *
+ * @property frequencyPenalty Frequency penalties.
+ *
+ * @property maxOutputTokens Specifies the maximum number of tokens that can be generated in the
+ * response. The number of tokens per word varies depending on the language outputted. Defaults to 0
+ * (unbounded).
+ *
+ * @property responseModality Specifies the format of the data in which the server responds to
+ * requests
+ *
+ * @property speechConfig Specifies the voice configuration of the audio response from the server.
+ *
+ * Refer to the
+ * [Control generated output](https://cloud.google.com/vertex-ai/generative-ai/docs/multimodal/control-generated-output)
+ * guide for more details.
+ */
+public class LiveGenerationConfig
+private constructor(
+  internal val temperature: Float?,
+  internal val topK: Int?,
+  internal val topP: Float?,
+  internal val candidateCount: Int?,
+  internal val maxOutputTokens: Int?,
+  internal val presencePenalty: Float?,
+  internal val frequencyPenalty: Float?,
+  internal val responseModality: ContentModality?,
```

Comment:
I don't think we should re-use [`ContentModality`](https://cloud.google.com/vertex-ai/generative-ai/docs/reference/rest/v1beta1/ModalityTokenCount#Modality) in `LiveGenerationConfig` since it's not the same proto as [`GenerationConfig.Modality`](https://cloud.google.com/vertex-ai/generative-ai/docs/reference/rest/v1beta1/GenerationConfig#Modality).

Reply:
Thanks for taking care of this!